### PR TITLE
 Handle non-existant config files correctly 

### PR DESCRIPTION
**Pull request by Emantor**
The code reinitializes the swayidle state, which clears the previously
parsed commands from the commandline. Remove this clear.

### CODE REVIEW

**Comment by Emantor** [comment]
@maringuu why did you clear the state when the config file is missing?

**Comment by maringuu** [comment]
Oh yeah, i frogot that case.

Let's say there is an error in the config at the second line.
The first line will be parsed and be saved in the state. I cleared the state so that the config will not be loaded at all.

Am 3. März 2020 19:26:25 MEZ schrieb Rouven Czerwinski <notifications@github.com>:

File: main.c [hunk 1]
```diff
@@ -783,8 +783,6 @@ int main(int argc, char *argv[]) {
 	}
 
 	if (load_config(config_path) == -1) {
```
Comment (emersion) [inline main.c]: We should probably hard-fail on config file error, and not error out if the config file doesn't exist.

**Comment by Emantor** [comment]
This should be ready to go.

File: main.c [hunk 2]
```diff
@@ -782,10 +782,13 @@ int main(int argc, char *argv[]) {
 		config_path = get_config_path();
 	}
 
-	if (load_config(config_path) == -1) {
-		swayidle_finish();
-		swayidle_init();
+	int config_load = load_config(config_path);
+
+	if (config_load == -ENOENT) {
 		swayidle_log(LOG_DEBUG, "Failed to load config. Starting without it...");
```
Comment (emersion) [inline main.c]: Can we update this message?

File: main.c [hunk 3]
```diff
@@ -782,10 +782,13 @@ int main(int argc, char *argv[]) {
...
+	} else if (config_load == -EINVAL){
```
Comment (emersion) [inline main.c]: Style: missing space between paren and bracket.

**Pull request merged**